proto/buf/alpha/registry/v1alpha1/doc.proto

```diff
@@ -274,6 +301,11 @@ message MessageField {
   }
 }
 
+// MessageOptions provides information for the documentation of options for a message.
+message MessageOptions {
+  bool deprecated = 1;
```

Comment:
What happens to a message if the _file_ is marked deprecated - will we also show it as deprecated in the UI?

Reply:
hm that's a good question. If a file is deprecated, I think it makes sense for every element defined inside it to be deprecated.

Reply:
I agree. 

This is a very similar situation to protobuf-es. In JavaScript, you can deprecate all kinds of elements, but not an entire file. That is why treat all elements in a deprecated file as _implicitly_ deprecated.

Reply:
Right. I guess that doesn't require a change in this proto structure, does it?
I'd need to look into how I can map `deprecated = true` for the `implicitly` deprecated elements in the business layer. OR, would it be better off just overwriting them as deprecated in the frontend if the file is deprecated?

Reply:
You have two options here:
1. In the BE, set the `deprecated` field for types whose parent file is deprecated
2. Change the proto structure to include information about file level deprecation, and add logic in the FE

Do you think there are any showstoppers or disadvantages with either option for what you want to do?

Reply:
The option 1 seems like the way to go, minimizing data transfer, unless we want to differentiate those `deprecated`  vs. `implicitly deprecated` elements rendered in the frontend. I'm not sure if its distinction would provide any values for consumers though. 🤔 What do you think?

Reply:
For option 2, consider the following proto file:

```proto
option deprecated = true;

message MyMessage {
  bool x = 1;
}
```

We would render it in bufdoc as:

## ~~MyMessage~~

```proto
message MyMessage {
  option deprecated = true;
  bool x = 1;
}
```

Do we _want_ to insert line 2, or do we only do that because we modeled the data structure in a way we have no other option? It doesn't seem correct to me to show something in the source that is not there.

Reply:
Thank you for the example. I see it now. It definitely doesn't look correct to have a deprecated line in the source, which would be inevitable if we don't distinguish them in the data structure. We'll have to provide information about file level deprecation.

Reply:
Added `FileOptions` to include `deprecated`